PhotoDetailScreen.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect } from 'react';
+import { StyleSheet, View, Text, Image, TouchableOpacity } from 'react-native';
+import { RouteProp, useRoute, useNavigation } from '@react-navigation/native';
+
+type RootStackParamList = {
+  PhotoDetail: { imageUrl: string };
+  ImageModal: { imageUrl: string };
+};
+
+type PhotoDetailRouteProp = RouteProp<RootStackParamList, 'PhotoDetail'>;
+
+const PhotoDetailScreen = () => {
+  const route = useRoute<PhotoDetailRouteProp>();
+  const imageUrl = route.params.imageUrl;
+  const navigation = useNavigation();
```

Comment:
navigation hook should be typed. I think I left the same comment in the last PR so I won't take anything for missing types here.